### PR TITLE
Fix drive classpath  issue for init_spark_on_yarn

### PR DESCRIPTION
**Pull request by jenniew**
Fix drive classpath  issue for init_spark_on_yarn.

Related issue: https://github.com/intel-analytics/analytics-zoo/issues/2531

### CODE REVIEW

File: pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def test_estimator_graph(estimator_for_spark_fixture):
 
     model = SimpleModel()
     file_path = os.path.join(resource_path, "orca/learn/ncf.csv")
-    data_shard = zoo.orca.data.pandas.read_csv(file_path, sc)
+    data_shard = zoo.orca.data.pandas.read_csv(file_path)
```
Comment (hkvision) [inline pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py:49]: e.... I forgot to change this when I remove RayXShards, but why jenkins doesn't report this error = =

File: pyzoo/zoo/common/nncontext.py [hunk 2]
```diff
@@ -338,11 +338,17 @@ def init_spark_conf(conf=None):
...
     spark_conf.setAll(zoo_conf.items())
 
     if os.environ.get("BIGDL_JARS", None) and not is_spark_below_2_2():
-        submit_args = " --driver-class-path " + os.environ["BIGDL_JARS"] + " pyspark-shell "
+        if 'PYSPARK_SUBMIT_ARGS' in os.environ:
+            submit_args = os.environ['PYSPARK_SUBMIT_ARGS']
+            start = submit_args.find('pyspark-shell')
+            submit_args = '{}--driver-class-path {} {}'.format(submit_args[:start],
+                                                               os.environ["BIGDL_JARS"],
+                                                               submit_args[start:])
```
Comment (hkvision) [inline pyzoo/zoo/common/nncontext.py:349]: why need this? why don't you directly append `--driver-class-path` to it?

File: pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py (hunk 1 above)
Comment (jenniew) [reply pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py:49]: I think we ignore all the test in /zoo/orca/learn/. We need to review tun-pytest to make necessary tests run.

File: pyzoo/zoo/common/nncontext.py (hunk 2 above)
Comment (jenniew) [reply pyzoo/zoo/common/nncontext.py:349]: I think we need to insert '--driver-class-path' before 'pyspark-shell'.

**Comment by jenniew** [comment]
Jenkins pass: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/3719/

File: pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py (hunk 1 above)
Comment (hkvision) [reply pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py:49]: mxnet tests are not ignored. Why are tf tests ignored?

Comment (jenniew) [reply pyzoo/test/zoo/orca/learn/spark/test_estimator_for_spark.py:49]: mxnet tests are in run-pytests-ray. So they can run.

**Pull request merged**